### PR TITLE
Add `dem_palette` as dataset

### PR DESCRIPTION
**Pull request by krpaulson**
## Describe changes

Setup `dem_palette` as an included dataset.

For discussion -- does this seem like an okay setup for this or should we try something more fancy per the links in #3 ?

## What issues are related

Related to #3 

### Packages Repositories

* [x] Have you read the [contributing guidelines](https://github.com/ihmeuw-demographics/packageTemplate/wiki#guide-to-r-package-development) for `ihmeuw-demographics` R packages?
* [x] Have you successfully run `devtools::check()` locally?
* [x] Have you updated or added function (and vignette if applicable) documentation? Did you update the 'man' and 'NAMESPACE' files with `devtools::document()`?
* [x] Have you added in tests for the changes included in the PR?
* [x] Do the changes follow the `ihmeuw-demographics` [code style](https://github.com/ihmeuw-demographics/packageTemplate/wiki/Code-style-guide)?
* [ ] Do your changes need to be immediately included in a new build of [`docker-base`](https://github.com/ihmeuw-demographics/docker-base) or [`docker-internal`](https://github.com/ihmeuw-demographics/docker-internal)? If so follow directions in those repositories to rebuild and redeploy the images.

### CODE REVIEW

File: data-raw/dem_palette.R [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+library(data.table)
+
+# set colors
+dem_palette <- c(
+  "gbd_previous_cycle" = "#e41a1c",  # red
+  "gbd_current"  = "#377eb8",        # blue
+  "gbd_recent"   = "#ff9a1a",        # orange
+  "vr"           = "#984ea3",        # purple
+  "hmd"          = "#b3adb4",        # grey
+  "wpp"          = "#4daf4a",        # green
```
Comment (chacalle) [inline data-raw/dem_palette.R:12]: How did you pick these colors? I think WPP for example would be good to make the blue from the top left logo "#00A1D9" on here https://population.un.org/wpp/

Like from the top left image here which I extracted using the Mac Digital Color Meter (command + shift + C)

File: data-raw/dem_palette.R [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+library(ggplot2)
+library(data.table)
+
+# set colors
+dem_palette <- c(
+  "gbd_previous_cycle" = "#e41a1c",  # red
+  "gbd_current"  = "#377eb8",        # blue
+  "gbd_recent"   = "#ff9a1a",        # orange
+  "vr"           = "#984ea3",        # purple
+  "hmd"          = "#b3adb4",        # grey
```
Comment (chacalle) [inline data-raw/dem_palette.R:11]: #02028B
https://www.mortality.org/

File: data-raw/dem_palette.R [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  "wpp"          = "#4daf4a",        # green
+  "stage_1"      = "#3cf837",        # lime green
+  "stage_2"      = "#3326c7",        # royal blue
+  "prior"        = "#8d8ae0",        # lavendar
+  "posterior"    = "#d99adb",        # pink
+  "data"         = "#292c3d"         # dark grey
```
Comment (chacalle) [inline data-raw/dem_palette.R:17]: Do we need different categories for VR and data? Should they be the same color

File: data-raw/dem_palette.R [hunk 4]
```diff
@@ -0,0 +1,33 @@
+
+library(ggplot2)
+library(data.table)
+
+# set colors
+dem_palette <- c(
+  "gbd_previous_cycle" = "#e41a1c",  # red
+  "gbd_current"  = "#377eb8",        # blue
```
Comment (chacalle) [inline data-raw/dem_palette.R:8]: #CBE19A, #57AF55, #373B74 How about these three colors from the IHME website? http://www.healthdata.org/

File: data-raw/dem_palette.R (hunk 1 above)
Comment (chacalle) [reply data-raw/dem_palette.R:12]: Also WPP recent and previous colours could be good

File: data-raw/dem_palette.R [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  "gbd_recent"   = "#ff9a1a",        # orange
+  "vr"           = "#984ea3",        # purple
+  "hmd"          = "#b3adb4",        # grey
+  "wpp"          = "#4daf4a",        # green
+  "stage_1"      = "#3cf837",        # lime green
+  "stage_2"      = "#3326c7",        # royal blue
+  "prior"        = "#8d8ae0",        # lavendar
+  "posterior"    = "#d99adb",        # pink
```
Comment (chacalle) [inline data-raw/dem_palette.R:16]: Should this be equivalent to GBD current?

File: R/data.R [hunk 6]
```diff
@@ -0,0 +1,22 @@
+#' @title Demographics palette
+#'
+#' @description Palette which can be used as `values` argument to
+#'   `ggplot2::scale_color_manual` or `ggplot2::scale_fill_manual`, and
+#'   includes common GBD comparators and output types.
+#'
+#' @format
+#' A named list, where the names are:
+#'   - "gbd_previous_cycle"
```
Comment (chacalle) [inline R/data.R:9]: Maybe a description of what these mean would be helpful

**Review by chacalle — APPROVED** [review]
I think this structure is good since those links seem to be for a bit different problem

File: data-raw/dem_palette.R [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+  "data"         = "#292c3d"         # dark grey
+)
+
```
Comment (chacalle) [inline data-raw/dem_palette.R:19]: How do we want to represent estimates before scaling?

File: data-raw/dem_palette.R (hunk 1 above)
Comment (krpaulson) [reply data-raw/dem_palette.R:12]: I picked the colors from the super pdfs where it made sense to, plus some random selection. Happy to make changes.

File: data-raw/dem_palette.R (hunk 3 above)
Comment (krpaulson) [reply data-raw/dem_palette.R:17]: Maybe data is one color and then data type is a shape?

File: data-raw/dem_palette.R (hunk 7 above)
Comment (krpaulson) [reply data-raw/dem_palette.R:21]: Added a `unscaled` entry. So for ST-GPR models you'd have stage1, stage2, unscaled (output from GPR), gbd current (final scaled results).

File: data-raw/dem_palette.R [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+library(ggplot2)
+library(data.table)
+
+# set colors
+dem_palette <- c(
+  "gbd_previous_cycle" = "#CBE19A",  # light green
+  "gbd_current"  = "#57AF55",        # ihme green
+  "gbd_recent"   = "#256b23",        # dark green
```
Comment (chacalle) [inline data-raw/dem_palette.R:9]: We need to switch the dark and light green to UW purple and gold to be able to distinguish better https://www.washington.edu/brand/graphic-elements/primary-color-palette/

**Comment by krpaulson** [comment]
image

**Comment by krpaulson** [comment]
Used the colors here to get UW purple and gold: https://globalhealth.washington.edu/communications

**Review by chacalle — APPROVED** [review]
Do we want to move this to `demUtils` and get rid of `demViz` if we aren't going to keep the plotting functions.

**Pull request merged**